backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionDAORdbImpl.java

```diff
@@ -60,12 +60,20 @@
           .append(ModelDBConstants.DATASET_ID)
           .append(" = :datasetId AND dsv.")
           .append(ModelDBConstants.VERSION)
-          .append(" =:version")
+          .append(" =:version ")
+          .append(" AND dsv.")
+          .append(ModelDBConstants.DELETED)
+          .append(" = false ")
+          .toString();
+  private static final String CHECK_DATASET_VERSION_EXISTS_BY_ID_HQL =
+      new StringBuilder("Select count(*) From DatasetVersionEntity dsv where ")
```

Comment:
Don't `count(*)`. Count on top of an index.

Reply:
I do not understand this comment. we have just got the count based on condition then what is the wrong here?

Reply:
if you do a count(id) on the table and there is an index then there is high chance that tidb will pick the right index.

Reply:
Ok make sense

Reply:
fixed

Reply:
We know `count(*)` is slow. We don't know why, but we get large processing time on hibernate doing that. We don't need to us to do it too if we have a reasonable alternative.